apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPoolOptions.cs

```diff
@@ -0,0 +1,233 @@
+        private TimeSpan _idleSessionRefreshDelay = TimeSpan.FromMinutes(15);
+        private TimeSpan _poolEvictionDelay = TimeSpan.FromDays(7);
+        private ResourcesExhaustedBehavior _waitOnResourcesExhausted = ResourcesExhaustedBehavior.Block;
+        private double _writeSessionsFraction;
+        private int _timeout = 60;
+        private TimeSpan _maintenanceLoopDelay = TimeSpan.FromSeconds(30);
+        private int _maximumConcurrentSessionCreates = 10;
+        private RetrySettings.IJitter _sessionRefreshJitter = new ProportionalRandomJitter(0.1);
+        private RetrySettings.IJitter _sessionEvictionJitter = new ProportionalRandomJitter(0.1);
+        private IClock _clock = SystemClock.Instance;
+        private IScheduler _scheduler= SystemScheduler.Instance;
+
+        /// <summary>
+        /// Constructs a new <see cref="SessionPoolOptions"/> with default values.
+        /// </summary>
+        public SessionPoolOptions()
+        {
+        }
+
+        /// <summary>
+        /// Maximum number of sessions that can be active per database. An active session is one that has been
+        /// acquired but not yet released back to the pool.
+        /// </summary>
+        /// <remarks>
+        /// This property has a minimum value of 1, and a default value of 100.
+        /// </remarks>
+        public int MaximumActiveSessions
+        {
+            get => _maximumActiveSessions;
+            set => _maximumActiveSessions = GaxPreconditions.CheckArgumentRange(value, nameof(value), 1, int.MaxValue);
+        }
+
+        /// <summary>
+        /// The minimum number of sessions to maintain in the pool of available sessions.
+        /// If the number of pooled sessions falls below this number, more sessions are added automatically.
+        /// </summary>
+        /// <remarks>
+        /// This property has a minimum value of 0, and a default value of 10.
+        /// </remarks>
+        public int MinimumPooledSessions
+        {
+            get => _minimumPooledSessions;
+            set => _minimumPooledSessions = GaxPreconditions.CheckNonNegative(value, nameof(value));
+        }
+
+        /// <summary>
+        /// The amount of time a session must be idle before it is refreshed.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This property must always be positive. The default value is 15 minutes.
+        /// </para>
+        /// <para>
+        /// The exact value used is subject to "jitter" to avoid a lot of sessions being refreshed at the exact same time.
+        /// </para>
+        /// <para>
+        /// A lower value will cause sessions to be refreshed more often, slightly reducing the risk of sessions expiring
+        /// while being used, at the cost of performing more refreshes.
+        /// </para>
+        /// <para>
+        /// This value must be less than the expire timer on the Spanner server which is currently set at 60 minutes.
```

Comment:
This is not being enforced. If the server value changes we should change the comment, so we could change the validation code as well I think. Although I understand there might not be a way to automatically detect changes on the server value and if it depends on us realizing the comment without the validation code might be OK really.

Reply:
Not only that - if we validate it here, we're forcing users to upgrade in order to use the new server settings. I definitely don't want to validate that here.